levels.go

```diff
@@ -14,6 +14,24 @@ const (
 	LevelFatal
 )
 
+// LevelFromString returns the level based on its string representation
+func LevelFromString(l string) Level {
+	switch l {
+	case "debug":
+		return LevelDebug
+	case "info":
+		return LevelInfo
+	case "warn":
+		return LevelWarn
+	case "error":
+		return LevelError
+	case "fatal":
+		return LevelFatal
+	default:
+		return LevelInfo
+	}
+}
+
```

Comment:
Added the implementation of both TextMarshaler and TextUnmarshaler in the last 2 commits